common/config/rush/common-versions.json

```diff
@@ -53,6 +53,12 @@
     "@azure/event-hubs": ["^2.1.4"],
 
     // Allow storage-blob-changefeed and storage-file-datalake to use the preview version of storage-blob.
-    "@azure/storage-blob": ["^12.2.0-preview.1"]
+    "@azure/storage-blob": ["^12.2.0-preview.1"],
+
```

Comment:
I've added a 2nd pass to lower the node types down to 10.x.

`@types/node`: These can be bumped down to 8, but it causes a build failure due to the OTel dependencies using v14 types (`Nodejs.Timer` vs `Nodejs.Timeout`) . I can investigate on OTel side to see if the types over there can be downgraded as well.

`@opentelemetry/api`: I would like the exporter to always be running tests on the latest version of OTel. Once `core-tracing` uses the latest OTel version, this exception can go away.

Reply:
My understanding is that packages should use the version of `@types/node` corresponding to the **lowest** supported Node version.  So if OTel supports Node 8 and above, then it should depend on `@types/node@^8.0.0`.  Otherwise, it's possible to use types which are available at build time but not run time.

@bterlson, @xirzec: Do you know if my statement above is correct?

Reply:
This is correct. The reasoning is that using Node 8 types ensures we won't use features that aren't present in later versions of node.

Reply:
In this case it appears that `@opentelemtry/api` is using the wrong version of `@types/node`:

```
"engines": {
  "node": ">=8.0.0"

"devDependencies": {
    "@types/node": "14.0.13",
```

https://unpkg.com/browse/@opentelemetry/api@0.9.0/package.json

Reply:
@mikeharder I've updated the comments in `common-versions.json` to describe when the `@types/node` exception can be removed (next OpenTelemetry release). 
https://github.com/Azure/azure-sdk-for-js/pull/10221/files#diff-a6b62ad391759bcd87094775766de9ffR58-R64

I think the `@opentelemetry/api` exception should stay permanently so the exporter can test on the latest OTel version without forcing the rest of this repo to also aggressively update.

WDYT?

Reply:
I approve the temporary addition of `@types/node`.

I defer to @xirzec regarding the version of `@opentelemetry/api`.  If there is a good reason different packages in this repo need to reference different versions of `@opentelemetry/api`, then this is the correct way to configure rush to allow this.  But it is usually considered a good thing for all packages in our repo to use the same dependency versions.

Reply:
@mikeharder I synced offline with @xirzec on `@opentelemetry/api`. He showed me the current cosmosdb approach to test on [additional typescript versions](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/cosmosdb/cosmos/consumer-test.js) via [`cosmos-additional-steps.yml`](https://github.com/Azure/azure-sdk-for-js/blob/b2732ee51c7a7efd7c9b33ff1f6220d0e07e62fa/eng/pipelines/templates/steps/cosmos-additional-steps.yml). Once `monitor-additional-steps.yml` is implemented, I can remove this `@opentelemetry/api` exception and test the exporter with additional OTel versions thru CI instead.

Overall, there is a clear path to remove both of these exceptions in the near-term.